server.js

```diff
@@ -23,3 +23,38 @@ app.get('/api/v1/all-products', async (request, response) => {
   }
 });
 
+
+// app.post('/api/v1/favorite-product', async (request, response) => {
+//   const product = request.body;
+
+//   if(!product.id) {
+//     response
+//       .status(422)
+//       .send({ error: 'Id is required.'});
+//   }
+
+//   try {
+//     const id = await database('product').insert(product, 'id');
+//     response.status(201).json({ id })
+//   } catch (error) {
+//     response.status(500).json({ error });
+//   }
+// });
+
+app.put('/api/v1/favorite/:id', async (request, response) => {
+  try {
+    const { id } = request.params;
+    const product = await database('product')
+      .where('id', '=', id)
+    const isFavorite = product[0].favorite;
+    await database('product')
+      .where('id', '=', id)
+      .update({favorite: !isFavorite});
+    response.status(201).json(product);
+  } catch (error) {
+    response.status(500);
+    console.error(error);
+  }
+});
+
+// app.post('/api/v1/favorite-product');
```

Comment:
What do you think about deleting this line since this comment is no longer needed?

Reply:
I agree, I will get this deleted!